tests/phpunit/tests/option/option.php

```diff
@@ -312,4 +312,103 @@ public function data_option_autoloading() {
 			array( 'autoload_false', false, 'no' ),
 		);
 	}
+
+	/**
+	 * @ticket 22192
+	 */
+	public function test_add_option_with_value_of_false_should_store_false_in_the_cache() {
+		add_option( 'foo', false );
+		$a = wp_cache_get( 'alloptions', 'options' );
+		$this->assertSame( false, $a['foo'] );
+	}
+
+	/**
+	 * @ticket 22192
+	 */
+	public function test_add_option_with_value_of_false_should_store_empty_string_in_the_database() {
+		add_option( 'foo', false );
+
+		// Delete cache to ensure we pull from the database.
+		wp_cache_delete( 'alloptions', 'options' );
+
+		$this->assertSame( '', get_option( 'foo' ) );
+	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @dataProvider data_update_option_type_juggling
```

Comment:
The parameters from a data provider should be documented in the test method's docblock.

:1234: Applies to a later test method too